File: exercise/java/day24/src/main/java/domain/core/Aggregate.java
package domain.core;

import io.vavr.collection.Seq;

import java.util.UUID;

public interface Aggregate {
    UUID getId();

    int getVersion();

    void applyEvent(Event event);

    Seq<Event> getUncommittedEvents();

    void clearUncommittedEvents();
}

File: exercise/python/day24/domain/core/aggregate.py
import uuid
from typing import List

from domain.core.event import Event


class Aggregate:
    def __init__(self):
        self.id = uuid.uuid4()
        self.version = 0
        self.uncommitted_events: List[Event] = []

    def apply_event(self, event: Event):
        raise NotImplementedError

    def get_uncommitted_events(self) -> List[Event]:
        return self.uncommitted_events

    def clear_uncommitted_events(self):
        self.uncommitted_events = []
